cluster/get-kube.sh

```diff
@@ -61,16 +61,24 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# TODO(pipejakob): Use dl.k8s.io/bazel once the redirection uses the correct bucket.
+KUBERNETES_BAZEL_RELEASE_URL="${KUBERNETES_BAZEL_RELEASE_URL:-${KUBERNETES_RELEASE_URL:-https://storage.googleapis.com/kubernetes-release-dev/bazel}}"
 # If KUBERNETES_RELEASE_URL is overridden but KUBERNETES_CI_RELEASE_URL is not then set KUBERNETES_CI_RELEASE_URL to KUBERNETES_RELEASE_URL.
 KUBERNETES_CI_RELEASE_URL="${KUBERNETES_CI_RELEASE_URL:-${KUBERNETES_RELEASE_URL:-https://dl.k8s.io/ci}}"
 KUBERNETES_RELEASE_URL="${KUBERNETES_RELEASE_URL:-https://dl.k8s.io}"
 
+KUBE_BRANCH_REGEX="[a-z0-9\-\_\.]+"
+KUBE_PULL_REQUEST_REGEX="[0-9]+"
+KUBE_COMMIT_REGEX="[a-f0-9]+"
+
 KUBE_RELEASE_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-([a-zA-Z0-9]+)\\.(0|[1-9][0-9]*))?$"
 KUBE_CI_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-([a-zA-Z0-9]+)\\.(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*)\\+[-0-9a-z]*)?$"
+KUBE_BAZEL_VERSION_REGEX="^bazel/((v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-([a-zA-Z0-9]+)\\.(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*)\\+[-0-9a-z]*)?)|(${KUBE_PULL_REQUEST_REGEX}/${KUBE_BRANCH_REGEX}:${KUBE_COMMIT_REGEX}(,${KUBE_PULL_REQUEST_REGEX}:${KUBE_COMMIT_REGEX})+))$"
```

Comment:
This seems like it has more to do with whether or not we use branch:commit,pull:commit or v1.2.3+commit, not whether or not it is bazel